PHYSOR26/post_processing.py

```diff
@@ -0,0 +1,122 @@
+import os
+import subprocess
+import pandas as pd
+
+"""
+A simple script that creates 
+    1. the union mesh
+    2. project the solution to that union mesh
+    3. Export element vertex average data to csv files  
+"""
+
+class post_processing_with_two_brain_cell:
+
+    def __init__(self,
+                 ref_dir_path:str,
+                 test_dir_path:str,
+                 variable:str):
+        
+        """
+        ref_dir_path: relative path of the ref dir solution 
+        test_dir_path: relative path of the test dir solution
+        variables_to_export: list of the variables to export in csv files 
+        """
+
+        self.ref_dir_path =  ref_dir_path 
+        self.test_dir_path =  test_dir_path
+        self.variable = variable
+
+        self.union_mesh_script = "common_mesh.i"
+        
+    def set_executable(self, executable_path):
+        """sets the absolute path of the executable"""
+        self.executable = executable_path
+
+    def generate_union_mesh(self, time_step):
+
+        if time_step == 1:
+            exodus_file_name = "openmc_out.e"
+        else:
+            exodus_file_name = f"openmc_out.e-s{time_step:03d}"
+
+        self.ref_mesh  = os.path.join(self.ref_dir_path, exodus_file_name)
+        self.test_mesh = os.path.join(self.test_dir_path, exodus_file_name)
+        
+        self.ref_test_mesh_arguments= [f"test_mesh_file_name={self.test_mesh}", f"ref_mesh_file_name={self.ref_mesh}"]
+        print (f"compairing betweem {self.ref_mesh} and {self.test_mesh}", )
+        try:
+            subprocess.run(
+                [self.executable, "-i", self.union_mesh_script, *self.ref_test_mesh_arguments, "--n-threads=28"],
+                cwd=os.getcwd(),
+                capture_output=True,
+                text=True,
+                check=True,
+            )
+
+
+            print(" ========== Union mesh generation is completed successfully! ==========")
+            self.set_updated_union_mesh()
+
+        except subprocess.CalledProcessError as e:
+            print("STDERR:", e.stderr)
+
+    def set_updated_union_mesh(self):
+
+        """
+        now that the union mesh is generated I need to set it.
+        I can iterate through the files in that dir and pick up the latest time step
+        """
+        prefix = self.union_mesh_script[:-2] + "_out.e-s00"
+        union_mesh_time_steps = [int(f[len(prefix):]) for f in os.listdir('.') if f.startswith(prefix) and f[len(prefix):].isdigit()]
```

Comment:
Can't you use pathlib and just get all the files with this prefix and take the last one because they'll be in alphabetical/numerical order?